L1Trigger/TrackFindingTracklet/plugins/ProducerKFin.cc

```diff
@@ -0,0 +1,261 @@
+    // ED output token for stubs
+    EDPutTokenT<TTDTC::Streams> edPutTokenAcceptedStubs_;
+    EDPutTokenT<TTDTC::Streams> edPutTokenLostStubs_;
+    // ED output token for tracks
+    EDPutTokenT<StreamsTrack> edPutTokenAcceptedTracks_;
+    EDPutTokenT<StreamsTrack> edPutTokenLostTracks_;
+    // Setup token
+    ESGetToken<Setup, SetupRcd> esGetTokenSetup_;
+    // DataFormats token
+    ESGetToken<DataFormats, DataFormatsRcd> esGetTokenDataFormats_;
+    // LayerEncoding token
+    ESGetToken<LayerEncoding, LayerEncodingRcd> esGetTokenLayerEncoding_;
+    // TrackBuilderChannel token
+    ESGetToken<TrackBuilderChannel, TrackBuilderChannelRcd> esGetTokenTrackBuilderChannel_;
+    // configuration
+    ParameterSet iConfig_;
+    // helper class to store configurations
+    const Setup* setup_;
+    // helper class to extract structured data from TTDTC::Frames
+    const DataFormats* dataFormats_;
+    // helper class to encode layer
+    const LayerEncoding* layerEncoding_;
+    // helper class to assign tracks to channel
+    TrackBuilderChannel* trackBuilderChannel_;
+    //
+    bool enableTruncation_;
+  };
+
+  ProducerKFin::ProducerKFin(const ParameterSet& iConfig) :
+    iConfig_(iConfig)
+  {
+    const InputTag& inputTag = iConfig.getParameter<InputTag>("InputTag");
+    const string& branchAcceptedStubs = iConfig.getParameter<string>("BranchAcceptedStubs");
+    const string& branchAcceptedTracks = iConfig.getParameter<string>("BranchAcceptedTracks");
+    const string& branchLostStubs = iConfig.getParameter<string>("BranchLostStubs");
+    const string& branchLostTracks = iConfig.getParameter<string>("BranchLostTracks");
+    // book in- and output ED products
+    edGetTokenTTTracks_ = consumes<TTTracks>(inputTag);
+    edPutTokenAcceptedStubs_ = produces<TTDTC::Streams>(branchAcceptedStubs);
+    edPutTokenAcceptedTracks_ = produces<StreamsTrack>(branchAcceptedTracks);
+    edPutTokenLostStubs_ = produces<TTDTC::Streams>(branchLostStubs);
+    edPutTokenLostTracks_ = produces<StreamsTrack>(branchLostTracks);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    esGetTokenLayerEncoding_ = esConsumes<LayerEncoding, LayerEncodingRcd, Transition::BeginRun>();
+    esGetTokenTrackBuilderChannel_ = esConsumes<TrackBuilderChannel, TrackBuilderChannelRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+    layerEncoding_ = nullptr;
+    trackBuilderChannel_ = nullptr;
+    //
+    enableTruncation_ = iConfig.getParameter<bool>("EnableTruncation");
+  }
+
+  void ProducerKFin::beginRun(const Run& iRun, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    if (!setup_->configurationSupported())
+      return;
+    // check process history if desired
```

Comment:
Change comment to "Check DTC cfg consistent with stub process history"

Reply:
that comment would be not correct, checkHistory is not constraint to DTC config or stub processing.

Reply:
Could the comment clarify why the process history is being checked? The comment for checkHistory() in TrackerDTC/Setup.h says it "checks current cfg vs input dataset cfg". I assume this means that it checks if the two are consistent? If not, I'm confused. Please clarify.